### ISSUE TITLE
[daikin] Thing status changes to OFFLINE (COMMUNICATION_ERROR) during getEnergyInfoYear() call

### ISSUE THREAD

**Issue by nepotu** [openhab/openhab-addons#8373]
I have two Daikin units controlled by BRP069A41 and BRP069A42. I noticed that often the devices do not react to commands. While checking the logs I noticed that the status of the device changes from ONLINE to OFFLINE (COMMUNICATION_ERROR).

I've checked the network and there are no issues on that side. Did also various curl loop tests with different sleep intervals and didn't notice anything abnormal. 

I don't think the thing should be marked as unavailable/offline if one API call specific to one feature fails.

## Current Behavior
events.log :
```
2020-08-31 14:48:37.272 [vent.ItemStateChangedEvent] - DaikinKB_OutTemp changed from 32.5 °C to 32.0 °C
2020-08-31 14:48:42.901 [hingStatusInfoChangedEvent] - 'daikin:ac_unit:daikin_mb_ac' changed from ONLINE to OFFLINE (COMMUNICATION_ERROR): Daikin HTTP error
2020-08-31 14:49:41.914 [hingStatusInfoChangedEvent] - 'daikin:ac_unit:daikin_mb_ac' changed from OFFLINE (COMMUNICATION_ERROR): Daikin HTTP error to ONLINE
```

openhab.log:
```
2020-08-31 14:49:41.901 [DEBUG] [n.internal.handler.DaikinBaseHandler] - Polling for state
2020-08-31 14:49:41.902 [DEBUG] [ing.daikin.internal.DaikinWebTargets] - Calling url: http://192.168.1.23/aircon/get_control_info
2020-08-31 14:49:41.913 [DEBUG] [ding.daikin.internal.api.ControlInfo] - Parsing string: "ret=OK,pow=1,mode=3,adv=,stemp=18.0,shum=0,dt1=25.0,dt2=M,dt3=18.0,dt4=25.0,dt5=25.0,dt7=25.0,dh1=AUTO,dh2=50,dh3=0,dh4=0,dh5=0,dh7=AUTO,dhh=50,b_mode=3,b_stemp=18.0,b_shum=0,alert=255,f_rate=A,f_dir=0,b_f_rate=A,b_f_dir=0,dfr1=5,dfr2=5,dfr3=A,dfr4=5,dfr5=5,dfr6=5,dfr7=5,dfrh=5,dfd1=0,dfd2=0,dfd3=0,dfd4=0,dfd5=0,dfd6=0,dfd7=0,dfdh=0"
2020-08-31 14:49:41.916 [DEBUG] [ing.daikin.internal.DaikinWebTargets] - Calling url: http://192.168.1.23/aircon/get_sensor_info
2020-08-31 14:49:41.922 [DEBUG] [nding.daikin.internal.api.SensorInfo] - Parsing string: "ret=OK,htemp=24.5,hhum=-,otemp=43.0,err=0,cmpfreq=16"
2020-08-31 14:49:41.923 [DEBUG] [ing.daikin.internal.DaikinWebTargets] - Calling url: http://192.168.1.23/aircon/get_year_power_ex
2020-08-31 14:49:53.311 [DEBUG] [internal.handler.DaikinAcUnitHandler] - getEnergyInfoYear() error: Daikin HTTP error
2020-08-31 14:49:56.924 [DEBUG] [internal.handler.DaikinAcUnitHandler] - getEnergyInfoYear() error: Daikin HTTP error
2020-08-31 14:50:37.892 [DEBUG] [scovery.DaikinACUnitDiscoveryService] - Starting broadcast for /172.17.255.255
2020-08-31 14:50:54.311 [DEBUG] [n.internal.handler.DaikinBaseHandler] - Polling for state
```

Seems that I get a `DaikinCommunicationException` while calling `getEnergyInfoYear()`. 

The curl result:
```
nepotu@e5440:~$ curl http://192.168.1.23/aircon/get_year_power_ex
ret=OK,curr_year_heat=0/0/0/0/0/0/0/0/0/0/0/0,prev_year_heat=0/0/0/0/0/0/0/0/0/0/0/0,curr_year_cool=0/0/0/0/0/0/0/0/0/0/0/0,prev_year_cool=0/0/0/0/0/0/0/0/0/0/0/0
```

## Your Environment
* Version used: openHAB 2.5.8 on docker (same behavior was noticed on the base OS as well)
* Operating System and version (desktop or mobile, Windows 10, Raspbian Buster, ...): Ubuntu 20.04.1 LTS

**Comment by jimtng** [comment]
pinging @LukasA83 (the developer who added the energy reading feature)

**Comment by rRobbieIT** [comment]
Same here

`2021-12-12 08:11:16.633 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'daikin:ac_unit:camera_ale_ac' changed from ONLINE to OFFLINE (COMMUNICATION_ERROR): Daikin HTTP error
2021-12-12 08:12:16.822 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'daikin:ac_unit:camera_ale_ac' changed from OFFLINE (COMMUNICATION_ERROR): Daikin HTTP error to ONLINE
2021-12-12 09:03:04.117 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'daikin:ac_unit:camera_ale_ac' changed from ONLINE to OFFLINE (COMMUNICATION_ERROR): Daikin HTTP error
2021-12-12 09:04:04.791 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'daikin:ac_unit:camera_ale_ac' changed from OFFLINE (COMMUNICATION_ERROR): Daikin HTTP error to ONLINE`

**Comment by jimtng** [comment]
@rRobbieIT is your issue related to getEnergyInfoYear() too?

**Comment by rRobbieIT** [comment]
@jimtng thank you for spending time in addressing the issue.

I believed so, but I am not sure anymore. Since I updated to OpenHAB 3.2.0 I found no error in the logs.

**Comment by jimtng** [comment]
@nepotu, I've just had a look into the daikin binding. The errors logged as `getEnergyInfoYear() error: Daikin HTTP error` is simply ignored. That alone shouldn't cause the thing to go offline.

I myself experience this issue of the thing going offline quite often and I'm using the Airbase controller, which doesn't make the call to getEnergyInfoYear(). 

I've submitted a PR #12239 that should hopefully fix this problem.

**Comment by jimtng** [comment]
@rRobbieIT thanks for the additional info. Could you please try the following jar and see if that fixed the issue for you?

https://github.com/jimtng/openhab-addons/releases/tag/daikin-test-v0.0.1

**Comment by rRobbieIT** [comment]
@jimtng many thanks for your help.

I have 1) uninstalled the existing binding Daikin from UI and then 2) installed and registered the jar you provided.

The binding is active and working.

`openhab> bundle:list -s | grep daikin
285 │ Active │  80 │ 3.3.0.202202091454    │ org.openhab.binding.daikin
openhab>`

I have been playing with the OpenHAB GUI since 15 minutes trying to replicate the error... so far so good. It did not show-up in the openhab logs. What it did show-up is this:

`roberto@vmserver8:~$ openhab-cli showlogs | grep Daikin
HeapByteBuffer@481542b6[p=0,l=3774,c=8192,r=3774]={<<<name":"Daikin Sala","note...ect_timestamp":16435872>>>_errors":..._mpdu_b}`

However both Daikin AC remained online, at least so it seems from the GUI.

![image](https://user-images.githubusercontent.com/78278792/153649455-0408c521-b38f-4332-8e0e-71868ad580ef.png)

I will be confirming in a couple of days how it is going.
Thanks

**Comment by jimtng** [comment]
> What it did show-up is this:
> 
> `roberto@vmserver8:~$ openhab-cli showlogs | grep Daikin HeapByteBuffer@481542b6[p=0,l=3774,c=8192,r=3774]={<<<name":"Daikin Sala","note...ect_timestamp":16435872>>>_errors":..._mpdu_b}`

Is the logging for org.openhab.binding.daikin set to DEBUG? Try changing that to INFO. The binding does some debug logging when it's doing retries.

**Comment by jimtng** [comment]
I've just uploaded an updated test version:
https://github.com/jimtng/openhab-addons/releases/tag/daikin-test-v0.0.2

**Comment by rRobbieIT** [comment]
@jimtng Just to give you an update: it is working fine,  thank you so much.

**Comment by jimtng** [comment]
@rRobbieIT thanks for the update. The fix has been merged today, so it should be in the next 3.3.0 snapshot.

**Comment by Villuendas** [comment]
Hi guys, I have the same problem.
I have 4 splits, 2 of them are new with built-in wifi adapters that work fine in Onecta.
The other 2 are old ones with BRP069B41's. These 2 go offline/online randomly very often in Onecta.

Can you please help me understand how to solve the issue?
I'm not a programmer so I would need some kind of step-by-step help if someone has the time to help me.

Thanks in advance!!

**Comment by rRobbieIT** [comment]
@Villuendas the issue has been corrected one year ago in OpenHAB version 3.3.0 and the disconnection did not appeared since then, now I am at version 3.4.2.

For any debug - and I am not a programmer too -  you should at least provide some logs information and OpenHAB version in use.

**Comment by rRobbieIT** [comment]
@Villuendas, 

Clear, I am afraid then this is not the right place to seek help on your issue which - apart form a similar effect - is not related in any shape or form to the Daikin tool.
